src/files/BrsFile.spec.ts

```diff
@@ -2837,4 +2882,225 @@ describe('BrsFile', () => {
+                };
+                expect(symbol.expandedTokenText).to.equal(lookup.name);
+                expect(symbol.type.equals(lookup.type, context)).be.true;
+            }
+        }
+
+        it('gets simple types based on the containing function expression', () => {
+            const file = program.setFile<BrsFile>('source/main.bs', `
+                sub doSomething(aInt as integer, aFloat as float, aStr as string, aBool as boolean, aObj as object)
+                    print aInt
+                    print aFloat
+                    print aStr
+                    print aBool
+                    print aObj
+                end sub
+            `);
+            const funcExpr = file.parser.references.functionExpressions[0];
+            const lookups = [
+                { line: 2, col: 28, name: 'aInt', type: new IntegerType() },
+                { line: 3, col: 28, name: 'aFloat', type: new FloatType() },
+                { line: 4, col: 28, name: 'aStr', type: new StringType() },
+                { line: 5, col: 28, name: 'aBool', type: new BooleanType() },
+                { line: 6, col: 28, name: 'aObj', type: new ObjectType() }
+            ];
+            checkSymbolLookups(file, funcExpr, lookups);
+            expectZeroDiagnostics(program);
+        });
+
+        it('gets custom types based on the containing function expression', () => {
+            const file = program.setFile<BrsFile>('source/main.bs', `
+                sub doSomething(klass as MyKlass)
+                    print klass
+                end sub
+
+                class MyKlass
+                end class
+            `);
+            const mainScope = program.getScopesForFile(file)[0];
+            mainScope.linkSymbolTable();
+            const funcExpr = file.parser.references.functionExpressions[0];
+            const token = file.parser.getTokenAt(Position.create(2, 28));
+            const symbol = file.getSymbolTypeFromToken(token, funcExpr, mainScope);
+            expect(symbol.expandedTokenText).to.equal('klass');
+            const classStmt = file.parser.references.classStatements[0];
+            expect(classStmt.name.text).to.equal('MyKlass');
+            expect(symbol.type.isAssignableTo(classStmt.getCustomType())).be.true;
+            expectZeroDiagnostics(program);
+        });
+
+        it('gets types of properties of a klass', () => {
+            const file = program.setFile<BrsFile>('source/main.bs', `
+                sub doSomething()
+                    klass = new MyKlass()
+                    print klass.name
+                    print klass.age
+                end sub
+
+                class MyKlass
+                    name as string
+                    age as integer
+                end class
+            `);
+            const funcExpr = file.parser.references.functionExpressions[0];
+            const lookups = [
+                { line: 3, col: 35, name: 'MyKlass.name', type: new StringType() },
```

Comment:
Since brighterscript is case-insensitive (and we're building our classes case insensitive), we should make sure these types of lookups work case-insensitive too. Can you add some tests to verify? Perhaps something like:
```
klass = new MyKlass()
print KLASS.NAME
print klass.AGE
```